File: 01_preseasonWeb/04_my_spaceship/ex00/my_spaceship.js
function my_spaceship(param_1) {
    length = param_1.length;
    cur_direct = 0;
    directions = ["up", "right", "down", "left"];
    var x = 0, y = 0;
    for (var i = 0; i < length; i++) {
        switch(param_1[i]) {
            case 'A':
                if (directions[cur_direct] == "up") {
                    y--;
                }
                else if (directions[cur_direct] == "right") {
                    x++;
                }
                else if (directions[cur_direct] == "down") {
                    y++;
                }
                else {
                    x--;
                }
                break;
            case 'R':
                cur_direct = (cur_direct + 1) % 4;
                break;
            case 'L':
                cur_direct = (--cur_direct < 0) ? 3 : cur_direct;
                break;
        }
    }
    return ("{x: " + x + ", y: " + y + ", direction: '" + directions[cur_direct] 
                    + "'}");  
}

File: 02_season01_arc01/01_bootcampC_arc1/04_quest04/ex05/my_spaceship.c
char* my_spaceship(char* param_1) {
    char* directions[] = {"up", "right", "down", "left"};
    int cur_dir = 0;
    int x = 0, y = 0;
    while(*param_1 != 0) {
        switch(*param_1) {
            case 'R':
                cur_dir = (cur_dir + 1) % 4;
                break;
            case 'L':
                cur_dir--;
                cur_dir = (cur_dir < 0) ? 3 : cur_dir;
                break;
            case 'A':
                switch (cur_dir) {
                    case 0:
                        y--;
                        break;
                    case 1:
                        x++;
                        break;
                    case 2:
                        y++;
                        break;
                    case 3:
                        x--;
                        break;
                    
                    default:
                        break;
                    }
                break;
        }
        param_1++;
    }
    //forming the output string
    char* res = (char*)malloc(100);
    sprintf(res, "{x: %d, y: %d, direction: '%s'}", x, y, directions[cur_dir]);
    return res;
}
